servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -521,7 +522,8 @@ private static StreamingHttpConnectionFilterFactory appendConnectionFilter(
     @Override
     public DefaultSingleAddressHttpClientBuilder<U, R> appendConnectionFactoryFilter(
             final ConnectionFactoryFilter<R, FilterableStreamingHttpConnection> factory) {
-        connectionFactoryFilter = connectionFactoryFilter.append(requireNonNull(factory));
+        connectionFactoryFilter = new ConnectionFactoryFilterAppender<>(connectionFactoryFilter,
+                requireNonNull(factory));
         strategyComputation.add(factory);
```

Comment:
Same here, the expectation is that `appendConnectionFactoryFilter` will be the only way to append them. Compare to the connection accepted case, this one already has `strategyComputation.add(factory)` that correctly computes the resulting strategy. We only need to sequence factories correctly.